riocli/device/delete.py

```diff
@@ -27,41 +36,143 @@
+        spinner.green.ok(Symbols.SUCCESS)
+        return
+
+    try:
+        devices = fetch_devices(
+            client, device_name_or_regex, delete_all)
+    except Exception as e:
+        spinner.text = click.style(
+            'Failed to delete device(s): {}'.format(e), Colors.RED)
+        spinner.red.fail(Symbols.ERROR)
+        raise SystemExit(1) from e
+
+    if not devices:
+        spinner.text = "No devices to delete"
+        spinner.ok(Symbols.SUCCESS)
+        return
+
+    headers = ['Name', 'Device ID', 'Status']
+    data = [[d.name, d.uuid, d.status] for d in devices]
+
     with spinner.hidden():
-        if not force:
-            click.confirm(
-                'Deleting device {} ({})'.format(
-                    device_name, device_guid), abort=True)
+        tabulate_data(data, headers)
+
+    spinner.write('')
+
+    if not force:
+        with spinner.hidden():
+            click.confirm('Do you want to delete above device(s)?',
+                          default=True, abort=True)
+        spinner.write('')
 
     try:
-        client = new_client(with_project=True)
-        handle_device_delete_error(client.delete_device(device_id=device_guid))
-        spinner.text = click.style('Device deleted successfully', fg=Colors.GREEN)
-        spinner.green.ok(Symbols.SUCCESS)
+        result = Queue()
+        func = functools.partial(_delete_deivce, client, result)
+        with ThreadPoolExecutor(max_workers=workers) as executor:
+            executor.map(func, devices)
+
+        result = sorted(list(result.queue), key=lambda x: x[0])
+
+        data, fg, statuses = [], Colors.GREEN, []
+        success_count, failed_count = 0, 0
+
+        for name, response in result:
+            if response.status_code and response.status_code < 400:
+                fg = Colors.GREEN
+                icon = Symbols.SUCCESS
+                success_count += 1
+            else:
+                fg = Colors.RED
+                icon = Symbols.ERROR
+                failed_count +=1
+                msg = get_error_message(response, name)
+                if msg:
+                    spinner.text = click.style(msg, Colors.YELLOW)
+                    spinner.ok(click.style(Symbols.WARNING, Colors.YELLOW))
+
+            data.append([
+                click.style(name, fg),
+                click.style(icon, fg)
+            ])
+
+        with spinner.hidden():
+            tabulate_data(data, headers=['Name', 'Status'])
+
+        spinner.write('')
+        if success_count:
+            spinner.text = click.style(
+                '{0} Devices(s) deleted successfully.'.format(success_count), Colors.GREEN)
+            spinner.ok(click.style(Symbols.SUCCESS, Colors.GREEN))
+        if failed_count:
+            spinner.text = click.style(
+                '{0} Devices(s) deletion failed.'.format(failed_count), Colors.YELLOW)
+            spinner.ok(click.style(Symbols.WARNING, Colors.YELLOW))
```

Comment:
Let's not do this. Instead if `failed_count` is 0 then show all devices deleted successfully. If `success_count` is 0 then show failed to delete devices. Else show `x/N devices deleted successfully`
```suggestion
        if success_count:
            spinner.text = click.style(
                '{0} Devices(s) deleted successfully.'.format(success_count), Colors.GREEN)
            spinner.ok(click.style(Symbols.SUCCESS, Colors.GREEN))
        if failed_count:
            spinner.text = click.style(
                '{0} Devices(s) deletion failed.'.format(failed_count), Colors.YELLOW)
            spinner.ok(click.style(Symbols.WARNING, Colors.YELLOW))
```